src/Sema.zig

```diff
@@ -12637,6 +12670,180 @@ fn zirReify(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Air.I
     }
 }
 
+fn reifyTuple(
+    sema: *Sema,
+    block: *Block,
+    src: LazySrcLoc,
+    fields_val: Value,
+) CompileError!Air.Inst.Ref {
+    const slice_val = fields_val.castTag(.slice).?.data;
+    const decl = slice_val.ptr.pointerDecl().?;
+    try sema.ensureDeclAnalyzed(decl);
```

Comment:
You're not guaranteed to have a `slice` tag, nor a `Decl` here. Just use `fields_val.slicePtr()` and `fields_val.sliceLen()`.

Note there is a handy `empty_struct_literal` for the case of empty tuple.